### ISSUE TITLE
Unable to make asynchronous requests

### ISSUE THREAD

**Issue by mhuggins** [n0mad01/node.bittrex.api#49]
I tried to wrap this API with promises so that I can use async/await.  I'm trying to hit the same API endpoint with different params in these calls.  Unfortunately, the same endpoint is hit repeatedly due to the `request_options` object being shared between calls.

**Comment by n0mad01** [comment]
can you describe that more precisely and maybe add a example?

**Comment by mhuggins** [comment]
Definitely, sorry for the lack of detail.

I've got a wrapper set up like this:

**client.js**
```js
import _ from "lodash";
import bittrex from "node.bittrex.api";

bittrex.options({ /* ... */ });

const FUNCTIONS = [
  "getmarkets",
  "getcurrencies",
  /* snip */
];

export default _.zipObject(FUNCTIONS, FUNCTIONS.map((fn) => {
  return (...args) => {
    return new Promise((resolve, reject) => {
      bittrex[fn](...args, (data) => {
        return data.success ? resolve(data.result) : reject(new Error(data.message));
      });
    });
  };
}));
```

All this does is wrap each function so that I can use the new `async`/`await` syntax, e.g.:

```js
import client from "./client";

async function getMarkets() {
  const markets = await client.getmarkets();
  console.log("markets:", markets);  // [{ BaseCurrency: "BTC", MarketCurrency: "BTC", ... }, ...]
}
```

The problem is now I'm trying to call the endpoint multiple times in a promise loop:

```js
const tickers = ["BTC-NEO", "BTC-ETH", /* ... */];

await Promise.all(tickers.map(async (ticker) => {
  const candles = await client.getcandles({ marketName: ticker, tickInterval: "oneMin" });
  await this.saveCandles(candles);
}));
```

Although the loop works properly, and I see that `getcandles` is called with iterating `marketName` values (i.e.: "BTC-NEO", then "BTC-ETH", etc.), the debug output shows that params never change for each "concurrent" request:

> requested from https://bittrex.com/Api/v2.0/pub/market/GetTicks?marketName=BTC-LTC&tickInterval=oneMin in: 0.539s
> requested from https://bittrex.com/Api/v2.0/pub/market/GetTicks?marketName=BTC-LTC&tickInterval=oneMin in: 0.539s
> requested from https://bittrex.com/Api/v2.0/pub/market/GetTicks?marketName=BTC-LTC&tickInterval=oneMin in: 0.539s
> requested from https://bittrex.com/Api/v2.0/pub/market/GetTicks?marketName=BTC-LTC&tickInterval=oneMin in: 0.539s

I'm not sure, but I think the problem is that within the node.bittrex.api package, `request_options` is shared across all calls.  I assume the package was not intended to be used in this way, but it would be convenient if it could be since I otherwise have to wait for one request has to complete before moving onto the next.  I'm trying to maximize speed and make my app efficient.

**Comment by mhuggins** [comment]
Thanks for merging my fix! 😄

**Comment by n0mad01** [comment]
thanks for the pull request - i also added you to the contributors list :)

**Comment by mhuggins** [comment]
Awesome, thanks!

**Comment by 4rocky4** [comment]
I think I have the same problem by using Q Library.

```
function test(coinList) {
  var deferred = Q.defer();
  var promises = [];
  coinList.forEach(function (coin) {
    promises.push(getCandles(coin));
  });

  return Q.allSettled(promises).then(function (results) {

    var listMarkets = [];
    results.forEach(function (r) {
      console.log(r.value.MarketName);
    });

    return listMarkets;
  });

}

function getCandles(coin) {
  var deferred = Q.defer();
  bittrex.getcandles({
    marketName: coin.MarketName,
    tickInterval: 'hour', // intervals are keywords (oneMin, fiveMin, hour, thirtyMin, Day)
  }, function (data, err) {
        if (err) {
          console.log('1 err:' + coin.MarketName);
          deferred.resolve(coin);
        }

        if (data) {
          console.log('1 :' + coin.MarketName);
          coin.Test = 10;
          deferred.resolve(coin);
        }
      });

  return deferred.promise;
}

```

- CoinList is a list of markets

It works but I get some errors at one time.
```
...
1 :BTC-NEOS
1 :BTC-LGD
1 :BTC-SPR
1 :BTC-XMG
1 :BTC-VTC
1 :BTC-QTUM
1 :BTC-MAID
1 :BTC-DOPE
1 :BTC-EXCL
1 :BTC-PAY
1 :BTC-NLG
1 :BTC-REP
1 err:BTC-OMG
1 err:BTC-GAM
1 err:BTC-KORE
1 err:BTC-SNT
1 err:BTC-NAV
1 err:BTC-EFL
1 err:BTC-EBST
1 err:BTC-PKB
```

I tried to call the same market with each request, I have the same problem.

Should I change something in my function to fix it ?
